smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/KafkaConnector.java

```diff
@@ -195,23 +196,34 @@ void init() {
             sources.add(source);
 
             boolean broadcast = ic.getBroadcast();
+            Multi<? extends Message<?>> stream;
+            if (!ic.getBatch()) {
+                stream = source.getStream();
+            } else {
+                stream = source.getBatchStream();
+            }
             if (broadcast) {
-                return ReactiveStreams.fromPublisher(source.getStream().broadcast().toAllSubscribers());
+                return ReactiveStreams.fromPublisher(stream.broadcast().toAllSubscribers());
             } else {
-                return ReactiveStreams.fromPublisher(source.getStream());
+                return ReactiveStreams.fromPublisher(stream);
             }
         }
 
         // create an instance of source per partitions.
-        List<Publisher<IncomingKafkaRecord<Object, Object>>> streams = new ArrayList<>();
+        List<Publisher<? extends Message<?>>> streams = new ArrayList<>();
         for (int i = 0; i < partitions; i++) {
             KafkaSource<Object, Object> source = new KafkaSource<>(vertx, group, ic, consumerRebalanceListeners,
                     kafkaCDIEvents, deserializationFailureHandlers, i);
             sources.add(source);
-            streams.add(source.getStream());
+            if (!ic.getBatch()) {
+                streams.add(source.getStream());
+            } else {
+                streams.add(source.getBatchStream());
+            }
         }
 
-        Multi<IncomingKafkaRecord<Object, Object>> multi = Multi.createBy().merging().streams(streams);
+        // TODO
```

Comment:
??

Reply:
I forgot to complete this TODO. I would suggest a `MultiMerge#streams(List<Publisher>)` in Mutiny.